Dockerfile

```diff
@@ -6,6 +6,7 @@ FROM ${BUILD_FROM}
 
 # Synchronize with homeassistant/core.py:async_stop
 ENV \
+    container=oci \
```

Comment:
This seems rather unused and trying to mimic something else.

It should be removed.

Reply:
It clearly conveys that a container is used. It's a convention used by other containers, see https://github.com/home-assistant/core/pull/132404#issuecomment-2528481195

Reply:
Can you point me to information about this convention?

Reply:
The linked comment / references are though, saying the opposite:

![CleanShot 2024-12-09 at 18 05 44](https://github.com/user-attachments/assets/0b1738aa-85c9-4d65-8d20-34ade43d784b)

As in, this environment var is supposed to be set by the upstream system (and not hardcoded in the image). As pointed out by the podman org member there.

> RedHat images happen to override by default, which is somewhat annoying.

With that in mind, I'm really hesitant to go down that same "annoying" path./

Reply:
I can see that.

Since the `is_official_image` check was removed per https://github.com/home-assistant/core/pull/132404#discussion_r1876365656 and now if we remove `container=dev` from the `Dockerfile`, this PR will no longer fix #127966 for Kubernetes users. Therefore, I suppose the next best thing would be to document (where would that best be documented?) what Kubernetes users should do, which is to set the `container=oci` environment variable (via helm chart or any other such mechanism).

Reply:
I added the check for `KUBERNETES_SERVICE_HOST` and I've removed the setting of `container=oci`.